cli/xml-to-ts.ts

```diff
@@ -0,0 +1,236 @@
+    .iterateNext();
+  iface.fields.push(...getInputFields(form));
+  iface.fields.push(...getFieldSetItems(form));
+  iface.fields.push(...getItemSetFields(form));
+  return iface;
+}
+
+function getInputFields(node): field[] {
+  const fields = [];
+  const inputs = xpath.evaluate(
+    "./input",
+    node,
+    null,
+    xpath.XPathResult.ANY_TYPE,
+    null
+  );
+  let input: XPathResult = null;
+  while ((input = inputs.iterateNext())) {
+    fields.push(createFieldFromInput(input));
+  }
+  return fields;
+}
+
+// FieldSet is a flat structure with no nesting.
+function getFieldSetItems(node): field[] {
+  const fields = [];
+  const fieldSets = xpath.evaluate(
+    "./field-set",
+    node,
+    null,
+    xpath.XPathResult.ANY_TYPE,
+    null
+  );
+
+  let fieldSet = null;
+  while ((fieldSet = fieldSets.iterateNext())) {
+    const items = xpath
+      .evaluate("./items", fieldSet, null, xpath.XPathResult.ANY_TYPE, null)
+      .iterateNext();
+    const inputs = xpath.evaluate(
+      "./input",
+      items,
+      null,
+      xpath.XPathResult.ANY_TYPE,
+      null
+    );
+
+    let input = null;
+    while ((input = inputs.iterateNext())) {
+      fields.push(createFieldFromInput(input));
+    }
+  }
+  return fields;
+}
+
+// ItemSet is a nested structure.
+function getItemSetFields(node): field[] {
+  const fields = [];
+  const itemSets = xpath.evaluate(
+    "./item-set",
+    node,
+    null,
+    xpath.XPathResult.ANY_TYPE,
+    null
+  );
+
+  let itemSet = null;
+  while ((itemSet = itemSets.iterateNext())) {
+    const name = xpath.select1("@name", itemSet).value;
+    const optional =
+      xpath.select1("./occurrences/@minimum", itemSet).value == 0;
+
+    const items = xpath
+      .evaluate("./items", itemSet, null, xpath.XPathResult.ANY_TYPE, null)
+      .iterateNext();
+    const subfields = [
+      ...getInputFields(items),
+      ...getFieldSetItems(items),
+      ...getItemSetFields(items)
+    ];
+
+    fields.push({ name, type: "Array", optional, subfields });
+  }
+  return fields;
+}
+
+/**
+ * Create a field from an <input> node.
+ * @param input an xpath input node
+ */
+function createFieldFromInput(input): field {
+  const name = xpath.select1("@name", input).value;
+  const comment = xpath.select1("string(./label)", input);
+  const optional = xpath.select1("./occurrences/@minimum", input).value == 0;
+  const type = "string";
+  return { name, type, comment, optional };
+}
+
+/**
+ * xmlToTs reads an xml file and returns a typescript-string.
+ * @param filename
+ */
+function xmlToTs(filename: string): string {
+  filename = path.basename(filename);
+  const xml = fs.readFileSync(filename, "utf-8");
+  const ifaceName = toPascalCase(
+    filename.substring(0, filename.lastIndexOf("."))
+  );
+  const iface = xmlToIface(ifaceName, xml);
+  return ifaceToTs(iface);
+}
+
+/**
+ * toPascalCase converts a string to PascalCase.
+ */
+function toPascalCase(str: string): string {
```

Comment:
Here I would probably just pull in an npm to do the job, instead of writing this myself:
https://www.npmjs.com/package/change-case